### PR TITLE
Fix EXPLAIN ANALYZE truncation

### LINKED ISSUE #4169: Long EXPLAIN ANALYZE plans can get truncated
The following explain analyze plan for a complex router query ends abruptly:
```sql
CREATE TABLE test (x int, y int);
SELECT create_distributed_table('test', 'x');
EXPLAIN (VERBOSE, ANALYZE) SELECT * FROM (SELECT x, count(*), row_number() OVER (), lead(x) OVER (PARTITION BY x) FROM test a JOIN test b USING (x) JOIN test c JOIN test d JOIN test e USING (x) USING (x)USING (x) JOIN test f USING (x) WHERE x = 2 OR x = 2 AND b.y > random() GROUP BY x ORDER BY x) foo ORDER BY 3 LIMIT 10;

...
│                                                         ->  Nested Loop  (cost=0.00..78.04 rows=121 width=4) (never executed)
│                                                               Output: d.x
│                                                               ->  Seq Scan on public.test_102065 d  (cost=0.00..38.25 rows=11 width=4) (never executed)
│                                                                     Output: d.x, d.y                                                                                            
│                                                                     Filter: (d.x = 2)
│                                                               ->  Materialize  (cost=0.00..38.30 rows=11 width=4) (never executed)
│                                                                     Output: e.x
│                                                                     ->  Seq Scan on public.test_102065 e  (cost=0.00..38.25 rows=11 width=4) (never executed)
│                                                                           Output: e.x
│                                                                           Filter: (e.x = 2)
│                                                         ->  Materialize  (cost=0.00..38.30 rows=11 widt
│ Planning Time: 1.720 ms
│ Execution Time: 6.746 ms
└─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
(58 rows)
```

### PR DESCRIPTION
**Pull request by marcocitus**
DESCRIPTION: Fix EXPLAIN ANALYZE truncation

We were truncating EXPLAIN ANALYZE output to RSIZE_MAX_STR via strlen_s. Use strnlen instead.

Fixes #4169

### CODE REVIEW

File: src/backend/distributed/planner/multi_explain.c [hunk 1]
```diff
@@ -1461,7 +1461,7 @@ WrapQueryForExplainAnalyze(const char *queryString, TupleDesc tupleDesc)
 static List *
 SplitString(const char *str, char delimiter)
 {
-	size_t len = strnlen_s(str, RSIZE_MAX_STR);
+	size_t len = strlen(str);
```
Comment (SaitTalhaNisanci) [inline src/backend/distributed/planner/multi_explain.c:1464]: Here I used `strnlen(str, RSIZE_MAX_MEM)` I guess there isn't a big difference here but using `strnlen` might be a bit safer maybe?

https://github.com/citusdata/citus/pull/4170/files#diff-44a470ebd7842955b493ca913941c03dR1464

**Comment by SaitTalhaNisanci** [comment]
This fixes #4169

File: src/backend/distributed/planner/multi_explain.c (hunk 1 above)
Comment (marcocitus) [reply src/backend/distributed/planner/multi_explain.c]: either way works for me, though given that this is a utility function I'll make it a parameter

File: src/test/regress/sql/multi_explain.sql [hunk 2]
```diff
@@ -1032,4 +1032,5 @@ PREPARE p4 (int, int) AS insert into users_table_2 ( value_1, user_id) select va
...
...000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000X';
```
Comment (SaitTalhaNisanci) [inline src/test/regress/sql/multi_explain.sql:1036]: Verified that with `master` we don't see `X` here.

File: src/backend/distributed/planner/multi_explain.c [hunk 3]
```diff
@@ -151,7 +151,7 @@ static void ExplainAnalyzeDestPutTuple(TupleDestination *self, Task *task,
 static TupleDesc ExplainAnalyzeDestTupleDescForQuery(TupleDestination *self, int
 													 queryNumber);
 static char * WrapQueryForExplainAnalyze(const char *queryString, TupleDesc tupleDesc);
-static List * SplitString(const char *str, char delimiter);
+static List * SplitString(const char *str, char delimiter, int maximumLength);
```
Comment (SaitTalhaNisanci) [inline src/backend/distributed/planner/multi_explain.c:154]: Very Minor (Feel free to ignore): `maxLength`

**Review by SaitTalhaNisanci — APPROVED** [review]
Could update `strlen -> strnlen` in the description.

**Pull request merged**